app/crud/category.py

```diff
@@ -0,0 +1,60 @@
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy import select, delete
+from uuid import UUID
+from typing import List, Optional
+
+from app.models.category import Category
+from app.schemas.category import CategoryCreate, CategoryUpdate
+
+
+async def create_category(db: AsyncSession, category_data: CategoryCreate) -> Category:
+    db_category = Category(
+        category_name=category_data.category_name,
+        category_color=category_data.category_color,
+    )
+    db.add(db_category)
+    await db.commit()
+    await db.refresh(db_category)
+    return db_category
+
+
+async def read_categories(db: AsyncSession) -> List[Category]:
+    result = await db.execute(select(Category))
+    return result.scalars().all()
+
+
+async def read_category_by_name(
+    db: AsyncSession, category_name: str
+) -> Optional[Category]:
+    result = await db.execute(
+        select(Category).where(Category.category_name == category_name)
+    )
+    return result.scalars().first()
+
+
+async def read_category_by_id(
+    db: AsyncSession, category_id: UUID
+) -> Optional[Category]:
+    result = await db.execute(select(Category).where(Category.id == category_id))
+    return result.scalars().first()
+
+
+async def update_category(
+    db: AsyncSession, category_id: UUID, category_data: CategoryUpdate
+) -> Category:
+    db_category = await read_category_by_id(db, category_id)
+
+    update_data = category_data.model_dump(exclude_unset=True)
+
+    for key, value in update_data.items():
+        setattr(db_category, key, value)
+
+    await db.commit()
+    await db.refresh(db_category)
+    return db_category
+
+
+async def delete_category(db: AsyncSession, category_id: UUID) -> bool:
+    await db.execute(delete(Category).where(Category.id == category_id))
+    await db.commit()
+    return True
```

Comment:
동작에 문제는 없으나 기존 코드들과의 일관성을 위해 아래처럼 고치면 좋을 것 같습니당

그리고 예외 처리는 위 코멘트가 받아들여지면 추가해주세용

```suggestion
    delete_statement = delete(Category).where(Category.id == category_id)

    try:
        await db.execute(delete_statement)
        await db.commit()
    except Exception as e:
        await db.rollback()
        raise e
    return True
```

Reply:
@GideokKim 
제 pr에도 남겨주셔서 저도 반영했습니다!

Reply:
네 반영완료했습니다